packages/commerce-server/src/format-prices-for-product.test.ts

```diff
@@ -484,34 +484,161 @@ test('PPP can be forced to not auto-apply for upgrade', async () => {
+      name: 'self paced workshop',
+      productType: 'self-paced',
     },
+  }
+
+  // @ts-ignore
+  mockCtx.prisma.product.findFirst.mockResolvedValueOnce({
+    ...mockAnotherSelfPacedProduct,
+  })
+
+  // @ts-ignore
+  mockCtx.prisma.price.findMany.mockResolvedValueOnce([
+    mockPriceAnotherSelfPaced,
+  ])
+
+  // @ts-ignore
+  mockCtx.prisma.purchase.findMany.mockResolvedValueOnce([
     {
-      status: 'Valid',
+      ...mockAnotherSelfPacedPurchase,
     },
-  ]
+  ])
 
   mockCtx.prisma.merchantCoupon.findMany.mockResolvedValue([MOCK_INDIA_COUPON])
 
   // @ts-ignore
-  mockCtx.prisma.purchase.findMany.mockResolvedValue(mockPurchases)
+  mockCtx.prisma.user.findFirst.mockResolvedValue({
+    id: 'user-1',
+  })
+
+  const {availableCoupons} = await formatPricesForProduct({
+    productId: DEFAULT_PRODUCT_ID,
+    userId: 'user-1',
+    quantity: 1,
+    country: 'IN',
+    ctx,
+  })
+
+  expect(availableCoupons.length).toBe(0)
+})
+
+test('PPP available for self-paced workshop having a valid live-event purchase', async () => {
+  const mockLiveEventProduct = {
+    id: 'product-live',
+    name: 'live event workshop',
+    createdAt: new Date(),
+    key: 'live event',
+    status: 1,
+    quantityAvailable: -1,
+    productType: 'live-event',
+  }
+
+  const mockPriceLiveEvent = {
+    id: 'price-live',
+    createdAt: new Date(),
+    status: 1,
+    productId: 'product-live',
+    nickname: 'live',
+    unitAmount: new Prisma.Decimal(55),
+  }
+
+  const mockLiveEventPurchase = {
+    totalAmount: new Prisma.Decimal(55),
+    id: 'purchase-live-event',
+    status: 'Valid',
+    createdAt: new Date(),
+    productId: 'product-live',
+    userId: 'user-1',
+    bulkCouponId: '',
+    city: '',
+    country: '',
+    ip_address: '',
+    merchantChargeId: '',
+    merchantSessionId: '',
+    redeemedBulkCouponId: '',
+    state: '',
+    upgradedFromId: '',
+    couponId: '',
+    product: {
+      id: 'product-live',
+      name: 'live event workshop',
+      productType: 'live-event',
+    },
+  }
+
+  // @ts-ignore
+  mockCtx.prisma.product.findFirst.mockResolvedValueOnce({
+    ...mockLiveEventProduct,
+    prices: [mockPriceLiveEvent],
+  })
+
+  // @ts-ignore
+  mockCtx.prisma.price.findMany.mockResolvedValueOnce([mockPriceLiveEvent])
+
+  // @ts-ignore
+  mockCtx.prisma.purchase.findMany.mockResolvedValueOnce([
+    {
+      ...mockLiveEventPurchase,
+    },
+  ])
+
+  mockCtx.prisma.merchantCoupon.findMany.mockResolvedValue([MOCK_INDIA_COUPON])
 
   // @ts-ignore
   mockCtx.prisma.user.findFirst.mockResolvedValue({
-    id: 'default-user',
+    id: 'user-1',
   })
 
-  const product = await formatPricesForProduct({
+  const {availableCoupons} = await formatPricesForProduct({
     productId: DEFAULT_PRODUCT_ID,
-    userId: 'default-user',
+    userId: 'user-1',
+    quantity: 1,
     country: 'IN',
     ctx,
   })
 
-  expect(product.availableCoupons.length).toBe(0)
+  expect(availableCoupons.length).toBe(1)
```

Comment:
this assertion could be strengthened to something like:

```suggestion
  expect(availableCoupons[0].type).toBe('ppp')
```